compiler/src/dmd/statementsem.d

```diff
@@ -1810,6 +1810,26 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
                 return setError();
             }
         }
+        else if (ps.ident == Id.mangle && ps._body)
```

Comment:
move the `ps._body` inside and give an appropriate error, otherwise it may give unrecognised pragma error, when it does recognise it but the user has not used it correctly.